### PR TITLE
Add integration models to linked spaces

### PR DESCRIPTION
**Pull request by stephherbers**
https://dimagi-dev.atlassian.net/secure/RapidBoard.jspa?rapidView=143&projectKey=USH&modal=detail&selectedIssue=USH-141

##### SUMMARY
This added Dialer, HMAC Callout, OTP passthrough to linked spaces so that their settings information can be entered once instead of 50+ times in some cases. I followed the steps in the [previous PR](https://github.com/dimagi/commcare-hq/pull/27855) for all three. For this PR, I pushed a commit for each of the steps outlined for just the dialer model, and for the other two just pushed when those steps were complete. Since they are a similar model, I assume that changes in one will require changes in all three, and that might help with following along. 

##### FEATURE FLAG
[Linked Project Spaces](https://confluence.dimagi.com/display/ccinternal/Linked+Project+Spaces?__hstc=187943799.4966953cf52b86b375bde4bc7705a59e.1600109107416.1600279776247.1600698112928.6&__hssc=187943799.2.1600698112928&__hsfp=1874026354)

##### RISK ASSESSMENT / QA PLAN
[QA Ticket](https://dimagi-dev.atlassian.net/browse/QA-1921?atlOrigin=eyJpIjoiNWE0YjliNzg0N2MwNDI3ZmI4NmUzNjljYTdmZWI5NDUiLCJwIjoiaiJ9)

### CODE REVIEW

File: corehq/apps/linked_domain/const.py [hunk 1]
```diff
@@ -22,6 +25,9 @@
     (MODEL_CASE_SEARCH, ugettext_lazy('Case Search Settings')),
     (MODEL_REPORT, ugettext_lazy('Report')),
     (MODEL_DATA_DICTIONARY, ugettext_lazy('Data Dictionary')),
+    (MODEL_DIALER_SETTINGS, ugettext_lazy('Dialer Settings')),
+    (MODEL_OTP_SETTINGS, ugettext_lazy('OTP Pass-through Settings')),
+    (MODEL_HMAC_CALLOUT_SETTINGS, ugettext_lazy('HMAC Callout Settings')),
```
Comment (orangejenny) [inline corehq/apps/linked_domain/const.py:30]: The UI for this calls it "Signed Callout" so that's the phrase to use here.

**Review by orangejenny — COMMENTED** [review]
This looks good.

One addition needed is to hide the new functionality from domains that don't have these feature flags turned on. If the flag is off, you shouldn't have the option to push/pull this data, and it also shouldn't show up in the Linked Project History report.

File: corehq/apps/linked_domain/local_accessors.py [hunk 2]
```diff
@@ -78,3 +79,35 @@ def get_data_dictionary(domain):
 
         data_dictionary[case_type.name] = entry
     return data_dictionary
+
+
+def get_dialer_settings(domain):
+    settings = DialerSettings.objects.get(domain=domain)
```
Comment (orangejenny) [inline corehq/apps/linked_domain/local_accessors.py:85]: What happens if you try to update these settings but the upstream domain doesn't have a `DialerSettings` object yet?

Comment (stephherbers) [reply corehq/apps/linked_domain/local_accessors.py]: fixed this issue (and with the other integration spaces) and they passed QA

File: corehq/apps/linked_domain/tasks.py [hunk 3]
```diff
@@ -160,8 +160,8 @@ def _release_report(self, domain_link, model):
                 text=_('Could not find report. Please check that the report has been linked.'),
             )
 
-    def _release_case_search(self, domain_link, model, user):
-        if not toggles.SYNC_SEARCH_CASE_CLAIM.enabled(domain_link.linked_domain):
+    def _release_flag_dependent_model(self, domain_link, model, user, feature_flag):
+        if not feature_flag.enabled(domain_link.linked_domain):
             return self._error_tuple(_("Case claim flag is not on"))
```
Comment (orangejenny) [inline corehq/apps/linked_domain/tasks.py:168]: This error message could be improved.

Comment (stephherbers) [reply corehq/apps/linked_domain/tasks.py:168]: good catch

**Review by orangejenny — APPROVED** [review]

**Pull request merged**